Failing ConcurrentSeqNoVersioningIT#testSeqNoCASLinearizability

testSeqNoCASLinearizability in ConcurrentSeqNoVersioningIT is failing with the current state of master:

```
java.lang.AssertionError: [zydtkfzqtheg.t][1], node[hcuRvnwrQIGXymYrYfaHbg], [R], s[STARTED], a[id=uQpPUT5GSCCPzLMYmrSmCQ] seq_no_stats mismatch
Expected: <SeqNoStats{maxSeqNo=7, localCheckpoint=7, globalCheckpoint=7}>
     but: was <SeqNoStats{maxSeqNo=7, localCheckpoint=7, globalCheckpoint=6}>
```

```
[ERROR] io.crate.integrationtests.disruption.seqno.ConcurrentSeqNoVersioningIT.testSeqNoCASLinearizability -- Time elapsed: 68.98 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 
[Must be linearizable] 
Expecting value to be true but was false
	at __randomizedtesting.SeedInfo.seed([8CB46E35E200B8C7:577A405FF8129A73]:0)
	at io.crate.integrationtests.disruption.seqno.ConcurrentSeqNoVersioningIT$Partition.assertLinearizable(ConcurrentSeqNoVersioningIT.java:505)
```

--- a/server/src/test/java/io/crate/integrationtests/disruption/seqno/ConcurrentSeqNoVersioningIT.java
+++ b/server/src/test/java/io/crate/integrationtests/disruption/seqno/ConcurrentSeqNoVersioningIT.java
@@ -479,7 +479,7 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
                         id, initialVersion, history);
             LinearizabilityChecker.SequentialSpec spec = new CASSequentialSpec(initialVersion);
 
-            boolean linearizable = false;
+            Boolean linearizable = null;
             try {
                 final ScheduledThreadPoolExecutor scheduler = Scheduler.initScheduler(Settings.EMPTY);
                 final AtomicBoolean abort = new AtomicBoolean();
@@ -496,13 +496,13 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
             } finally {
                 // implicitly test that we can serialize all histories.
                 String serializedHistory = base64Serialize(history);
-                if (linearizable == false) {
+                if (Boolean.TRUE.equals(linearizable) == false) {
                     // we dump base64 encoded data, since the nature of this test is that it does not reproduce even with same seed.
                     logger.error("Linearizability check failed for id: {}. Spec: {}, initial version: {}, serialized history: {}",
                                  id, spec, initialVersion, serializedHistory);
+                    assertThat(linearizable).as("Must not be unlinearizable").isNull();
                 }
             }
-            assertThat(linearizable).as("Must be linearizable").isTrue();
         }
     }
 

